### PR TITLE
sync: core repo master

### PR DESCRIPTION
**Pull request by tamtamchik**
This PR brings all the changes from the core repo.

TODO:

- [x] Port changes from https://github.com/lidofinance/lido-dao/issues/826 ⇒ separated into #845 
- [x] Apply changes from the latest core renovation patch https://github.com/lidofinance/core/pull/225

### CODE REVIEW

File: bugbounty.md [hunk 1]
```diff
@@ -1,11 +0,0 @@
-# Bug Bounties with Immunefi
```
Comment (failingtwice) [inline bugbounty.md]: I think we should leave this file

File: deployed-mainnet.json [hunk 2]
```diff
@@ -1,610 +0,0 @@
-{
```
Comment (failingtwice) [inline deployed-mainnet.json]: Not sure if we can remove deployed.json files yet

**Review by failingtwice — DISMISSED** [review]

File: deployed-mainnet.json (hunk 2 above)
Comment (tamtamchik) [reply deployed-mainnet.json]: Agree, let's leave them for now.

File: bugbounty.md (hunk 1 above)
Comment (tamtamchik) [reply bugbounty.md]: Will do, thank you!

Comment (tamtamchik) [reply bugbounty.md]: Restored in 99f378b0ce4d79f95c0d773e9db5513cc524522d

File: deployed-mainnet.json (hunk 2 above)
Comment (tamtamchik) [reply deployed-mainnet.json]: Fixed in e0cac15

**Review by failingtwice — DISMISSED** [review]
Thank you!

File: .solcover.js [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
-  }
-}
+  istanbulReporter: ["html", "text", "cobertura"],
+  // NOTE: those is are all relative paths to the `contracts` directory and does not support regex
+  //       https://github.com/sc-forks/solidity-coverage/issues/632#issuecomment-1736629543
+  skipFiles: [
+    "common/interfaces",
+    "common/test_helpers",
+    "0.4.24/template",
+    "0.4.24/test_helpers",
```
Comment (TheDZhon) [inline .solcover.js]: seems not relevant anymore

File: .solcover.js [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
-  mocha: {
-    enableTimeouts: false
-  }
-}
+  istanbulReporter: ["html", "text", "cobertura"],
+  // NOTE: those is are all relative paths to the `contracts` directory and does not support regex
+  //       https://github.com/sc-forks/solidity-coverage/issues/632#issuecomment-1736629543
+  skipFiles: [
+    "common/interfaces",
+    "common/test_helpers",
```
Comment (TheDZhon) [inline .solcover.js]: and this

File: .solcover.js [hunk 5]
```diff
@@ -1,7 +1,19 @@
...
+    "0.4.24/template",
+    "0.4.24/test_helpers",
+    "0.6.11/deposit_contract.sol",
+    "0.6.12/interfaces",
+    "0.6.12/mocks",
+    "0.8.9/interfaces",
+    "0.8.9/test_helpers",
```
Comment (TheDZhon) [inline .solcover.js]: and this

**Review by TheDZhon — DISMISSED** [review]
image

File: .solcover.js (hunk 3 above)
Comment (tamtamchik) [reply .solcover.js]: Fixed in https://github.com/lidofinance/lido-dao/pull/843/commits/57cd3d141f6ec69297f040454f7d388dc93e5351

File: .solcover.js (hunk 4 above)
Comment (tamtamchik) [reply .solcover.js]: Fixed in https://github.com/lidofinance/lido-dao/pull/843/commits/57cd3d141f6ec69297f040454f7d388dc93e5351

File: .solcover.js (hunk 5 above)
Comment (tamtamchik) [reply .solcover.js]: Fixed in https://github.com/lidofinance/lido-dao/pull/843/commits/57cd3d141f6ec69297f040454f7d388dc93e5351

**Review by TheDZhon — APPROVED** [review]

**Pull request merged**